WalletWasabi/KeyManagement/KeyManager.cs

```diff
@@ -495,19 +495,28 @@ public void SetPasswordVerified()
 
 		public IEnumerable<HdPubKey> GetKeys(KeyState? keyState = null, bool? isInternal = null)
 		{
-			if (keyState is null && isInternal is null)
+			if (keyState is null)
 			{
-				return GetKeys(x => true);
-			}
-			if (isInternal is null && keyState != null)
-			{
-				return GetKeys(x => x.KeyState == keyState);
+				if (isInternal is null)
+				{
+					return GetKeys(x => true);
+				}
+				else
+				{
+					return GetKeys(x => x.IsInternal == isInternal);
+				}
 			}
-			else if (keyState is null)
+			else
 			{
-				return GetKeys(x => x.IsInternal == isInternal);
+				if (isInternal is null)
+				{
+					return GetKeys(x => x.KeyState == keyState);
+				}
+				else
+				{
+					return GetKeys(x => x.IsInternal == isInternal && x.KeyState == keyState);
+				}
```

Comment:
The original one is better readable.

Reply:
Same here, there is redundant checks in the original code, in some cases you check 3 or 4 times.
E.g. if `keyState is null` and `isInternal != null` we need to check both of them two times in order to execute `return GetKeys(x => x.IsInternal == isInternal);` but in the proposed code you check each one of the only once.

Reply:
Yes, I've misread the changes. ACK for this, too.